tools/integration-tests/tester/tests/testutil.go

```diff
@@ -164,6 +179,49 @@ func CheckForMessageIDs(t *testing.T, peers []*framework.Peer, messageIDs map[st
 	}
 }
 
+// AwaitMessageAvailability awaits until the given message IDs become available on all given peers or
+// the max duration is reached. Returns a map of missing messages per peer. An error is returned if at least
+// one peer does not have all specified messages available.
+func AwaitMessageAvailability(peers []*framework.Peer, messageIDs map[string]DataMessageSent, waitFor time.Duration) (missing map[identity.ID]map[string]types.Empty, err error) {
+	s := time.Now()
+	missing = map[identity.ID]map[string]types.Empty{}
+
+	for ; time.Since(s) < waitFor; time.Sleep(500 * time.Millisecond) {
+		var wg sync.WaitGroup
+		wg.Add(len(peers))
+
+		for _, p := range peers {
+			go func(p *framework.Peer) {
+				defer wg.Done()
```

Comment:
We can add a check here to avoid requesting messages if a node has them all already. But still need a lock there.
```suggestion
			go func(p *framework.Peer) {
				defer wg.Done()
				
				// TODO: lock !
				if _, has := missing[p.ID()]; !has {
				    return
				} 
```